EmpleoDotNet.AppServices/JobOpportunityService.cs

```diff
@@ -8,14 +9,59 @@
 namespace EmpleoDotNet.AppServices
 {
     public class JobOpportunityService : IJobOpportunityService
-    {
+    {        
         public void CreateNewJobOpportunity(JobOpportunity jobOpportunity, string userid)
         {
             jobOpportunity.UserProfile = _userProfileRepository.GetByUserId(userid);
             _jobOpportunityRepository.Add(jobOpportunity);
             _jobOpportunityRepository.SaveChanges();
         }
 
+        public void UpdateJobOpportunity(int id, JobOpportunity updatedJob)
+        {
+            var existingJob = _jobOpportunityRepository.GetJobOpportunityById(id);
+
+            existingJob.Title = updatedJob.Title;
+            existingJob.Approved = updatedJob.Approved;
+            existingJob.Category = updatedJob.Category;
+            existingJob.CompanyEmail = updatedJob.CompanyEmail;
+            existingJob.CompanyLogoUrl = updatedJob.CompanyLogoUrl;
+            existingJob.CompanyName = updatedJob.CompanyName;
+            existingJob.CompanyUrl = updatedJob.CompanyUrl;
+            existingJob.Description = updatedJob.Description;
+            existingJob.HowToApply = updatedJob.HowToApply;
+            existingJob.IsActive = updatedJob.IsActive;
+            existingJob.IsRemote = updatedJob.IsRemote;
+
+            if (existingJob.JobOpportunityLocation != null)
```

Comment:
I thought I had solved this bug, sorry.

The point is to create an instance of JobOpportunityLocation in case doesn't exists (and needs to be updated) otherwise, update the existing instance.

Thanks for this. I'll submit a fix later.